server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/UnifiedHighlighter.java

```diff
@@ -62,20 +65,33 @@ public HighlightField highlight(HighlighterContext highlighterContext) {
         SearchContext context = highlighterContext.context;
         FetchSubPhase.HitContext hitContext = highlighterContext.hitContext;
         Encoder encoder = field.fieldOptions().encoder().equals("html") ? HighlightUtils.Encoders.HTML : HighlightUtils.Encoders.DEFAULT;
-        CustomPassageFormatter passageFormatter = new CustomPassageFormatter(field.fieldOptions().preTags()[0],
-            field.fieldOptions().postTags()[0], encoder);
+        PassageFormatter passageFormatter = new CustomPassageFormatter(field.fieldOptions().preTags()[0],
+                field.fieldOptions().postTags()[0], encoder);       
+
         final int maxAnalyzedOffset = context.indexShard().indexSettings().getHighlightMaxAnalyzedOffset();
 
         List<Snippet> snippets = new ArrayList<>();
         int numberOfFragments;
         try {
 
-            final Analyzer analyzer =
+            Analyzer analyzer =
                 getAnalyzer(context.mapperService().documentMapper(hitContext.hit().getType()), fieldType);
             List<Object> fieldValues = HighlightUtils.loadFieldValues(field, fieldType, context, hitContext);
+
             fieldValues = fieldValues.stream()
                 .map((s) -> convertFieldValue(fieldType, s))
                 .collect(Collectors.toList());
+            
+            //Annotated text fields are special - the highlighter's formatter and break iterators
+            // need to work with text minus markup and any re-analysis of the content requires the
+            // marked up versions of field values. This special analyzer helps deal with this.
+            if (fieldType.typeName().equals (AnnotatedTextFieldMapper.CONTENT_TYPE)){
+                AnnotatedHighlighterAnalyzer annotatedHighlighter = new AnnotatedHighlighterAnalyzer(fieldValues, analyzer);
+                fieldValues = annotatedHighlighter.getPlainTextValuesForHighlighter();
+                // use the wrapped analyzer that holds marked-up text
```

Comment:
@romseygeek Not sure what you've got cooking with new highlighters but I made need to avoid reusing TextFieldMapper's "phrase" field type and introduce a fork "annotations_phrase" because that's a similar special-case scenario like the plain "annotations" type. Before we feed the original JSON content into the highlighter's passage fragmentation logic we must strip the text of annotation markup using getPlainTextValuesForHighlighter